python/main.py

```diff
@@ -1,32 +1,80 @@
 import os
 import logging
 import pathlib
-from fastapi import FastAPI, Form, HTTPException
+from fastapi import FastAPI, Form, HTTPException, File, UploadFile
 from fastapi.responses import FileResponse
 from fastapi.middleware.cors import CORSMiddleware
+from typing import Optional
+import json
+import hashlib
 
 app = FastAPI()
 logger = logging.getLogger("uvicorn")
 logger.level = logging.INFO
 images = pathlib.Path(__file__).parent.resolve() / "images"
-origins = [ os.environ.get('FRONT_URL', 'http://localhost:3000') ]
+origins = [ os.environ.get('FRONT_URL', 'http://localhost:9000') ]
 app.add_middleware(
     CORSMiddleware,
     allow_origins=origins,
     allow_credentials=False,
     allow_methods=["GET","POST","PUT","DELETE"],
     allow_headers=["*"],
 )
+item_id_num = 1  # Initialize the item ID counter
 
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
+def add_item(name: Optional[str] = Form(None), category: Optional[str] = Form(None), image: UploadFile = File(...)):
+    # Load existing items from the file
+    with open("items.json") as file:
+        data = json.load(file)
+
+    # Access the global item ID counter
+    global item_id_num
+
+    # hash image using sha256
+    imageContent = image.file.read()
+    hashedImage = hashlib.sha256(imageContent).hexdigest()
+    hashedImg = hashedImage + os.path.splitext(image.filename)[1]
+
+    # Create a new item dictionary
+    new_item = {"id": item_id_num,"name": name, "category": category, "imageFile": hashedImg}
+
+    # Increment the item ID counter for the next item
+    item_id_num += 1
```

Comment:
not sure if this is better, but since the json file contents will be read, maybe it would be more reliable to check the id of the latest item and add 1?